x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/DeleteConnectorAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class DeleteConnectorAction extends ActionType<AcknowledgedResponse> {
+
+    public static final DeleteConnectorAction INSTANCE = new DeleteConnectorAction();
+    public static final String NAME = "cluster:admin/xpack/connector/delete";
```

Comment:
Question: Are we having a new cluster admin privilege for these actions?

Reply:
This is not a system index. Do we need any extra privileges? I was able to execute delete actions locally

Reply:
What are the permissions that a user need to have in order to manage connectors? Just permissions over the connectors index?

We probably want from a security / user roles perspective to limit those actions to specific users. These should probably be included as part of a `manage_connectors` cluster privilege.

Otherwise, we leave the option for non-authorized users to manage connectors and syncs.

Reply:
@carlosdelest Thank you for explaining, let me discuss this with the team. Do you think that adding the specific cluster privileges can be added in a separate PR? I need to figure out how to do this correctly.

Reply:
Thank you for sharing context, if it's indeed a small change I will try to squeeze this before the FF :)